// Copyright ICR
// 
// note: 
// cd <path>/rpc/cpp
// ../script/protoc.sh `pwd`/.. cpp `pwd`/../horos.proto `pwd`
// cd ../python
// ../script/protoc.sh `pwd`/.. python `pwd`/../horos.proto `pwd`

syntax = "proto3";

package icr;

// A simple key-value storage service
service Horos {
  // Request current dicom image
  rpc GetCurrentImageFile (DicomNameRequest) returns (DicomNameResponse) {}
  rpc GetCurrentImage (DicomImageRequest) returns (DicomImageResponse) {}
}

// The request message containing the key
message DicomNameRequest {
  string id = 1;
}

// The response message containing the value associated with the key
message DicomNameResponse {
  string dicom_name = 1;
}

message DicomImageRequest {
  string id = 1;
}

message DicomImageResponse {
  string dicom_name = 1;
  repeated int32 image_size = 2;
  repeated float voxel_size = 3;
  repeated float origin = 4;
  repeated float data = 5;
}
